src/scripts/DataLayerListenerManagerFactory.js

```diff
@@ -97,11 +98,17 @@ ListenerManagerFactory.create = function(dataLayerManager) {
       if (_isMatching(listener, item)) {
         const listenerConfig = listener.config;
         const itemConfig = item.config;
-        const itemConfigCopy = JSON.parse(JSON.stringify(itemConfig));
-        if (listener.config.selector) {
-          const oldValue = get(dataLayerManager._state, listenerConfig.selector);
-          const newValue = get(itemConfig.data, listenerConfig.selector);
-          listenerConfig.handler(itemConfigCopy, oldValue, newValue);
+        const itemConfigCopy = merge({}, itemConfig);
+        if (item.config.data) {
+          if (listener.config.selector) {
+            const oldValue = get(dataLayerManager._state, listenerConfig.selector);
+            const newValue = get(itemConfig.data, listenerConfig.selector);
+            listenerConfig.handler(itemConfigCopy, oldValue, newValue);
```

Comment:
@jckautzmann we could also call the handler with the data layer here:

`listenerConfig.handler.call(dataLayerManager._dataLayer, itemConfigCopy, oldValue, newValue);`